sensor/kubernetes/listener/resources/deployments.go

```diff
@@ -180,19 +177,42 @@ func (d *deploymentHandler) processWithType(obj, oldObj interface{}, action cent
 		d.endpointManager.OnDeploymentRemove(deploymentWrap)
 		d.processFilter.Delete(deploymentWrap.GetId())
 	}
-	if err := deploymentWrap.updateHash(); err != nil {
-		log.Errorf("UNEXPECTED: could not calculate hash of deployment %s: %v", deploymentWrap.GetId(), err)
-	}
 
 	events = d.appendIntegrationsOnCredentials(action, deploymentWrap.GetContainers(), events)
-	outputMessage := component.NewResourceEvent([]*central.SensorEvent{deploymentWrap.toEvent(action)}, []component.CompatibilityDetectionMessage{
-		{
-			Object: deploymentWrap.GetDeployment(),
-			Action: action,
-		},
-	}, nil)
 
-	events = component.MergeResourceEvents(events, outputMessage)
+	if features.ResyncDisabled.Enabled() {
+		if action == central.ResourceAction_REMOVE_RESOURCE {
+			// if resource is being removed, we can create the remove message here without related resources
+			events = component.MergeResourceEvents(events, component.NewResourceEvent([]*central.SensorEvent{deploymentWrap.toEvent(action)}, nil, nil))
```

Comment:
I think we need to append the ID here too. The detector needs to process the `remove events` in order to remove them from the deduper and send the empty alerts message to mark all previous alerts as stale.

Reply:
Fair enough. But I think we should use the `CompatibilityDetectionMessage` property for this. 

If we pass the deployment reference, it's a problem if the deployment was deleted from the internal store at the end of the handler. Then the resolver component won't be able to trigger the detection for a deployment that no longer exists.  

I think a definite solution for this should do one of the following:

1) When it's a remove event, copy the deployment object (doesn't matter if there are no related resources), send the entire object to the pipeline. This object is forwarded to the detection and ignored by the dependency resolution step. 
2) The detection package receives "delete" operations based on IDs. Not sure how possible is that, but if it's just removing from the deduper and sending an empty alert, it should be ok. 

My proposal is to do (1) for now, using the compatibility property. If we stick to it, this property will no longer be there only for compatibility reasons since we have a legitimate reason to send a copy of the object over the pipeline. 

If you're ok with (1), I'll update the code.

Reply:
By the way, there should be a test that fails if the empty alerts are not sent for a deployment that was deleted. 😞